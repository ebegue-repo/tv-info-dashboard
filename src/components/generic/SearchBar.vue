<template>  
  <div class="flex justify-center">
    <div class="mb-3 xl:w-96">
      <div class="relative mb-4 flex w-full flex-wrap items-stretch">
        <input
          type="search"
          v-model="searchInput"
          class="relative m-0 -mr-px block w-[1%] min-w-0 flex-auto rounded-l border border-solid border-neutral-300 bg-transparent bg-clip-padding px-3 py-1.5 text-base font-normal text-neutral-700 outline-none transition duration-300 ease-in-out focus:border-primary focus:text-neutral-700 focus:shadow-te-primary focus:outline-none dark:text-neutral-200 dark:placeholder:text-neutral-200"
          placeholder="Search"
          aria-label="Search"
          aria-describedby="button-addon1" />
        <custom-button class="ml-5" title="Search" positiveButton="true" @click="onButtonClick(searchInput)"/>
      </div>
    </div>
  </div>   
</template>

<script lang="ts">
    import CustomButton from '@/components/generic/CustomButton.vue';

    export default {
      name: 'SearchBar',
      components: {
        CustomButton,
      },
      methods: {
        onButtonClick(searchInput: string) {
          this.$emit('searchEvent', searchInput);
        },
      },
      data() {
        return {
          searchInput: '',
        }
      }
    }
</script>