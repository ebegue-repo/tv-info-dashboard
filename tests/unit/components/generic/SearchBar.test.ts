import { mount } from '@vue/test-utils';
import SearchBar from '@/components/generic/SearchBar.vue';
import { describe, expect, test, vi } from 'vitest';

describe('SearchBar', () => {
    test('It should render the generic search bar and match the snapshot', () => {
        const wrapper = mount(SearchBar);
        expect(wrapper.html()).toMatchSnapshot();
    });
});