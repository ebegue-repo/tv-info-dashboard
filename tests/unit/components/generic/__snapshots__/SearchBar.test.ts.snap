// Vitest Snapshot v1

exports[`SearchBar > It should render the generic search bar and match the snapshot 1`] = `
"<div class=\\"flex justify-center\\">
  <div class=\\"mb-3 xl:w-96\\">
    <div class=\\"relative mb-4 flex w-full flex-wrap items-stretch\\"><input type=\\"search\\" class=\\"relative m-0 -mr-px block w-[1%] min-w-0 flex-auto rounded-l border border-solid border-neutral-300 bg-transparent bg-clip-padding px-3 py-1.5 text-base font-normal text-neutral-700 outline-none transition duration-300 ease-in-out focus:border-primary focus:text-neutral-700 focus:shadow-te-primary focus:outline-none dark:text-neutral-200 dark:placeholder:text-neutral-200\\" placeholder=\\"Search\\" aria-label=\\"Search\\" aria-describedby=\\"button-addon1\\"><button type=\\"submit\\" class=\\"bg-tvm-green text-white inline-flex justify-center rounded-md py-2 px-5 mr-3 text-sm font-medium shadow-sm focus:outline-none focus:ring-2 focus:ring-indigo-500 focus:ring-offset-2 ml-5\\">Search</button></div>
  </div>
</div>"
`;
